packages/sdk/test/e2e/executeQuote.test.ts

```diff
@@ -0,0 +1,226 @@
+import { spokePoolAbi } from "../../src/abis/SpokePool";
+
+const inputToken = {
+  address: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+  symbol: "USDC",
+  name: "USD Coin",
+  decimals: 6,
+  logoUrl:
+    "https://raw.githubusercontent.com/across-protocol/frontend/master/src/assets/token-logos/usdc.svg",
+} as const;
+
+const inputAmount = 100;
+const inputAmountBN = parseUnits(inputAmount.toString(), inputToken.decimals);
+
+// MAINNET => ARBITRUM
+const testRoute = {
+  originChainId: 1,
+  destinationChainId: 42161,
+  originToken: inputToken.address,
+} as const;
+
+let route: Route;
+let quote: Quote;
+
+let approvalSimulationSuccess = false;
+let approvalTxSuccess = false;
+let depositSimulationSuccess = false;
+let depositTxSuccess = false;
+let fillTxSuccess = false;
+let fillHash: Hash | undefined;
+let fillLog: FilledV3RelayEvent | undefined;
+
+describe("executeQuote", async () => {
+  test("Gets available routes for intent", async () => {
+    const [_route] = await testClient.getAvailableRoutes(testRoute);
+    assert(_route !== undefined, "route is not defined");
+    assertType<Route>(_route);
+    route = _route;
+  });
+
+  test("Gets a quote for a route", async () => {
+    const _quote = await testClient.getQuote({
+      route,
+      inputAmount: inputAmountBN,
+    });
+
+    assert(_quote, "No quote for route");
+    assertType<Quote>(_quote);
+    quote = _quote;
+  });
+
+  describe("Executes a quote", async () => {
+    afterAll(async () => {
+      await Promise.all([
+        chainClientArbitrum.resetFork(),
+        chainClientMainnet.resetFork(),
+      ]);
+    });
+    beforeAll(async () => {
+      // sanity check that we have fresh anvil instances running in this case
+      const blockNumberArbitrum = await chainClientArbitrum.getBlockNumber();
+      expect(blockNumberArbitrum).toBe(BLOCK_NUMBER_ARBITRUM);
+      const blockNumberMainnet = await chainClientMainnet.getBlockNumber();
+      expect(blockNumberMainnet).toBe(BLOCK_NUMBER_MAINNET);
+
+      // fund test client and wallet clients with 1 ETH
+      await Promise.all([
+        chainClientMainnet.setBalance({
+          address: chainClientMainnet.account.address,
+          value: parseEther("1"),
+        }),
+        chainClientArbitrum.setBalance({
+          address: chainClientArbitrum.account.address,
+          value: parseEther("1"),
+        }),
+        chainClientMainnet.setBalance({
+          address: testWalletMainnet.account.address,
+          value: parseEther("1"),
+        }),
+      ]);
+
+      // fund test wallet clients with 1000 USDC
+      await fundUsdc(chainClientMainnet, testWalletMainnet.account.address);
+
+      const latestBlock = await chainClientMainnet.getBlock({
+        blockTag: "latest",
+      });
+
+      const depositTimestamp = latestBlock.timestamp - 20n;
+
+      // override quote timestamp
+      const deposit = {
+        ...quote.deposit,
+        quoteTimestamp: Number(depositTimestamp),
```

Comment:
Did you find that this was necessary? It should actually be possible to use `latestBlock.timestamp`.

Reply:
Ah it does actually work you're right.